tests/suite/test_custom_annotations.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+from settings import TEST_DATA, DEPLOYMENTS
+from suite.fixtures import PublicEndpoint
+from suite.resources_utils import create_items_from_yaml, delete_items_from_yaml, \
+    create_example_app, delete_common_app, wait_until_all_pods_are_ready, replace_configmap_from_yaml, \
+    get_ingress_nginx_template_conf, get_first_pod_name
+from suite.yaml_utils import get_first_ingress_host_from_yaml, get_names_from_yaml
+
+
+class CustomAnnotationsSetup:
+    """
+    Encapsulate Custom Annotations Example details.
+
+    Attributes:
+        public_endpoint (PublicEndpoint):
+        namespace (str):
+        ingress_host (str):
+    """
+    def __init__(self, public_endpoint: PublicEndpoint, ingress_name, namespace, ingress_host, ic_pod_name):
+        self.public_endpoint = public_endpoint
+        self.namespace = namespace
+        self.ingress_name = ingress_name
+        self.ingress_host = ingress_host
+        self.ic_pod_name = ic_pod_name
+        self.backend_1_url = f"http://{public_endpoint.public_ip}:{public_endpoint.port}/backend1"
+        self.backend_2_url = f"http://{public_endpoint.public_ip}:{public_endpoint.port}/backend2"
+
+
+@pytest.fixture(scope="class")
+def custom_annotations_setup(request, kube_apis, ingress_controller_prerequisites,
+                             ingress_controller_endpoint, test_namespace) -> CustomAnnotationsSetup:
+    ing_type = request.param
+    print("------------------------- Deploy ConfigMap with custom template -----------------------------------")
+    replace_configmap_from_yaml(kube_apis.v1,
+                                ingress_controller_prerequisites.config_map['metadata']['name'],
+                                ingress_controller_prerequisites.namespace,
+                                f"{TEST_DATA}/custom-annotations/{ing_type}/nginx-config.yaml")
+    print("------------------------- Deploy Custom Annotations Example -----------------------------------")
+    ing_src = f"{TEST_DATA}/custom-annotations/{ing_type}/annotations-ingress.yaml"
+    create_items_from_yaml(kube_apis, ing_src, test_namespace)
+    host = get_first_ingress_host_from_yaml(ing_src)
+    ingress_name = get_names_from_yaml(ing_src)[0]
+    create_example_app(kube_apis, "simple", test_namespace)
```

Comment:
do we need to deploy the backends for this? it seems that we only test the generated config

Reply:
It's not necessary indeed.